packit_service/worker/whitelist.py

```diff
@@ -0,0 +1,202 @@
+        result["status"] = str(result["status"])
+        return result
+
+
+class Whitelist:
+    def __init__(self):
+        self.db = PersistentDict(hash_name="whitelist")
+
+    @staticmethod
+    def _is_packager(account_login: str) -> bool:
+        """
+        If GitHub username is same as FAS username this method checks if user is packager.
+        User is considered to be packager when he/she has the badge:
+         `If you build it... (Koji Success I)`
+        :param account_login: str, Github username
+        :return: bool
+        """
+
+        url = f"https://badges.fedoraproject.org/user/{account_login}/json"
+        data = requests.get(url)
+        if not data:
+            return False
+        assertions = data.json().get("assertions")
+        if not assertions:
+            return False
+        for item in data.json().get("assertions"):
+            if "Succesfully completed a koji build." in item.get("description"):
+                logger.info(f"User: {account_login} is a packager in Fedora!")
+                return True
+        logger.info(
+            f"Cannot verify whether user: {account_login} is a packager in Fedora."
+        )
+        return False
+
+    def add_account(self, github_app: GithubAppData) -> bool:
+        """
+        Add account to whitelist, if automatic verification of user
+        (check if user is packager in fedora) fails, account is still inserted in whitelist
+         with status : `waiting`.
+         Then a scripts in files/scripts have to be executed for manual approval
+        :param github_app: github app installation info
+        :return:
+        """
+        # we want to verify if user who installed the application is packager
+        if Whitelist._is_packager(github_app.sender_login):
+            github_app.status = WhitelistStatus.approved_automatically
+            self.db[github_app.account_login] = github_app.get_dict()
+            logger.info(f"Account {github_app.account_login} whitelisted!")
+            return True
+        else:
+            logger.error(
+                "Failed to verify that user is Fedora packager. "
+                "This could be caused by different github username than FAS username "
+                "or that user is not a packager."
+            )
+            github_app.status = WhitelistStatus.waiting
+            self.db[github_app.account_login] = github_app.get_dict()
+            logger.info(
+                f"Account {github_app.account_login} inserted "
+                f"to whitelist with status: waiting for approval"
+            )
+            logger.info(self.db)
+            return False
+
+    def approve_account(self, account_name: str) -> bool:
+        """
+        Approve user manually
+        :param account_name: account name for approval
+        :return:
+        """
+        if account_name in self.db:
+            account = self.db[account_name]
```

Comment:
how about doing this right away and return false for keyerror?

Reply:
Personally, I do not like catching exceptions if it is not needed.

What about this one:

```python
if account_name not in self.db:
    return False
```

Reply:
my point is that what I'm proposing is more efficient: with this implementation we are hitting redis twice

Reply:
OK, that's a valid reason.

I've thought you were primary mentioning this:

```python
if something:
    # some work
    return True
else:
    return False
```
:arrow_down: 
```python
if not something:
    return False
# some work
return True
```